File: 680_ValidPalindrome2/validPalindrome2.js
var validPalindrome = function(s) {
        var isValidPalindrome = function(leftPtr, rightPtr) {
        while (leftPtr < rightPtr) {
            if (s[leftPtr] != s[rightPtr]) {
                return false
            }
            leftPtr++
            rightPtr--
        }    
        return true
    };
    
    let leftPtr = 0
    let rightPtr = s.length - 1

    while (leftPtr < rightPtr) {
        if (s[leftPtr] != s[rightPtr]) {
            return isValidPalindrome(leftPtr + 1,rightPtr) ||
                    isValidPalindrome(leftPtr, rightPtr - 1)
        }
        leftPtr++
        rightPtr--
        
    }

    return true
};

console.log(validPalindrome("aba"))
console.log(validPalindrome("abca"))
console.log(validPalindrome("abc"))
console.log(validPalindrome("abccbda"))

File: 680_ValidPalindrome2/validPalindrome2.py

def validPalindrome(s: str) -> bool:
    def isValidPalindrome(leftPtr: int, rightPtr: int) -> bool:
        while leftPtr < rightPtr:
            if s[leftPtr] != s[rightPtr]:
                return False
            leftPtr += 1 
            rightPtr -= 1  
        return True

    leftPtr = 0
    rightPtr = len(s) - 1

    while leftPtr < rightPtr:
        if s[leftPtr] != s[rightPtr]:
            return isValidPalindrome(leftPtr + 1, rightPtr) or isValidPalindrome(leftPtr, rightPtr - 1)
        leftPtr += 1
        rightPtr -= 1
    
    return True

print(validPalindrome("aba"))
print(validPalindrome("abca"))
print(validPalindrome("abc"))
print(validPalindrome("abccbda"))
        